### ISSUE TITLE
Support request bodies for GET requests?

### ISSUE THREAD

**Issue by gjdanis** [tiangolo/fastapi#1691]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
import uvicorn
from pydantic import BaseModel
from fastapi import FastAPI


class Item(BaseModel):
    name: str
    description: str


app = FastAPI()


@app.get("/items")
def create_item(item: Item):
    return item


if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=5000, debug=True)

```

You can run and test the above code with this CURL request:

```bash
curl -X GET \
  http://localhost:5000/items \
  -d '{
	"name": "test",
	"description": "tester"
}
'
```

### Description

FastAPI's documentation seems to suggest that GET requests with a body aren't supported: https://fastapi.tiangolo.com/tutorial/body/

However, running the above example it looks like this is supported, although the auto generated documentation misses documenting the parameters of the body.

Can this be supported in FastAPI? Our use case is that we need to support a GET method that could contain a sizable amount of data in the request. We don't really want to make this a POST method as the endpoint is only for retrieval and isn't modifying any data. Everything appears to be working for us except the auto generated documentation. 

### Environment

* OS: [e.g. Linux / Windows / macOS]: Windows
* FastAPI Version [e.g. 0.3.0]: 0.54.1
* Python Version: 3.8.2

**Comment by Kludex** [comment]
I don't recall it properly, but I saw an issue talking about it... It seems like Swagger didn't support previously, but they are going to support/or are already supporting. There's an open PR related to what you're talking about, you might want to check: https://github.com/tiangolo/fastapi/pull/1626/

**Comment by tiangolo** [comment]
Yeah, that's currently supported by FastAPI in recent versions (the current is `0.62.0`), but discouraged, as it's not part of the web standards. And the docs with Swagger UI will probably not show the optional GET body, but curl should work.

**Comment by terminalmage** [comment]
I have kind of the opposite issue, in that I have not defined a request body for my GET requests, but swagger is prepopulating it with:

```
[
    "string"
]
```

This forces me to always remove the request body before executing, when running any API calls through the swagger UI.

I've searched for how to hide this, but the only examples I've found so far are for Java applications.
